src/main/java/org/folio/consortia/utils/TenantContextUtils.java

```diff
@@ -34,6 +34,14 @@ public static FolioExecutionContext createFolioExecutionContext(MessageHeaders h
     return getContextFromKafkaHeaders(headers, moduleMetadata, centralTenantId);
   }
 
+  public static FolioExecutionContext prepareContextForTenant(String tenantId, FolioModuleMetadata folioModuleMetadata, FolioExecutionContext context) {
+    if (MapUtils.isNotEmpty(context.getOkapiHeaders())) {
+      context.getOkapiHeaders().put(XOkapiHeaders.TENANT, List.of(tenantId));
+      log.info("FOLIO context initialized with tenant {}", tenantId);
+    }
+    return new DefaultFolioExecutionContext(folioModuleMetadata, context.getOkapiHeaders());
+  }
+
```

Comment:
another method getFolioExecutionContextCopyForTenant() used only in tests now, correct? if yes please delete it

Reply:
by the way, we already have this method in TenantContextUtils, can we use it?
createFolioExecutionContextForTenant